moveit_core/collision_detection_bullet/src/collision_robot_bt.cpp

```diff
@@ -0,0 +1,354 @@
+        fcl_obj.collision_objects_.push_back(
+            FCLCollisionObjectPtr(new fcl::CollisionObjectd(objs[k]->collision_geometry_, fcl_tf)));
+        // we copy the shared ptr to the CollisionGeometryData, as this is not stored by the class itself,
+        // and would be destroyed when objs goes out of scope.
+        fcl_obj.collision_geometry_.push_back(objs[k]);
+      }
+  }
+}
+
+void CollisionRobotBt::allocSelfCollisionBroadPhase(const robot_state::RobotState& state, FCLManager& manager) const
+{
+  auto m = new fcl::DynamicAABBTreeCollisionManagerd();
+  // m->tree_init_level = 2;
+  manager.manager_.reset(m);
+  constructFCLObject(state, manager.object_);
+  manager.object_.registerTo(manager.manager_.get());
+  // manager.manager_->update();
+}
+
+void CollisionRobotBt::checkSelfCollision(const CollisionRequest& req, CollisionResult& res,
+                                          const robot_state::RobotState& state) const
+{
+  checkSelfCollisionHelper(req, res, state, nullptr);
+}
+
+void CollisionRobotBt::checkSelfCollision(const CollisionRequest& req, CollisionResult& res,
+                                          const robot_state::RobotState& state, const AllowedCollisionMatrix& acm) const
+{
+  checkSelfCollisionHelper(req, res, state, &acm);
+}
+
+void CollisionRobotBt::checkSelfCollision(const CollisionRequest& req, CollisionResult& res,
+                                          const robot_state::RobotState& state1,
+                                          const robot_state::RobotState& state2) const
+{
+  ROS_ERROR_NAMED("collision_detection.bullet", "FCL continuous collision checking not yet implemented");
+}
+
+void CollisionRobotBt::checkSelfCollision(const CollisionRequest& req, CollisionResult& res,
+                                          const robot_state::RobotState& state1, const robot_state::RobotState& state2,
+                                          const AllowedCollisionMatrix& acm) const
+{
+  ROS_ERROR_NAMED("collision_detection.bullet", "FCL continuous collision checking not yet implemented");
+}
+
+void CollisionRobotBt::checkSelfCollisionHelper(const CollisionRequest& req, CollisionResult& res,
+                                                const robot_state::RobotState& state,
+                                                const AllowedCollisionMatrix* acm) const
+{
+  // updating the link position with the current robot state
+  for (std::size_t i = 0; i < geoms_.size(); ++i)
+    if (geoms_[i] && geoms_[i]->collision_geometry_)
+    {
+      bt_manager_.setCollisionObjectsTransform(
+          geoms_[i]->collision_geometry_data_->ptr.link->getName(),
+          state.getCollisionBodyTransform(geoms_[i]->collision_geometry_data_->ptr.link,
+                                          geoms_[i]->collision_geometry_data_->shape_index));
+    }
+
+  tesseract::ContactResultMap contact_map;
+  bt_manager_.contactTest(contact_map, tesseract::ContactTestType::FIRST, acm, req);
```

Comment:
This line causes an error when compiling: the forked tesseract you have only takes the ACM as the third argument, and not the request. Did you push your latest commits there?

Reply:
Ah sorry, you are right, forgot to push my latest changes!

Reply:
I pushed them now.